### DISCUSSION TITLE
Quasar in production - Refresh - 404 Not Found

### DISCUSSION THREAD

**Discussion by Cabezagithub** [quasarframework/quasar#14910] [CLI - PWA mode] [answered]
I have a Quasar SPA in production and when I try to refresh (F5) I get the following error: 404 Not Found.
The SPA is deployed on Ubuntu 20.04 with Nginx.
Build the SPA on Windows 11. I compiled with Quasar build and uploaded the files. Running "Quasar dev" on my Notebook I have no problems with refresh (F5), everything works fine.
The first installation in production had Apache2 installed, I stopped it and installed Nginx but the problem persists, so I think it must be a Quasar configuration that I don't have correctly.
It's driving me crazy... Any suggestions?

My development Quasar info:
Operating System - Windows_NT(10.0.25217) - win32/x64
NodeJs - 14.17.3

Global packages
  NPM - 6.14.13
  yarn - Not installed
  @quasar/cli - 1.3.2
  @quasar/icongenie - Not installed
  cordova - Not installed

Important local packages
  quasar - 2.10.2 -- Build high-performance VueJS user interfaces (SPA, PWA, SSR, Mobile and Desktop) in record time
  @quasar/app-webpack - 3.6.2 -- Quasar Framework App CLI with Webpack
  @quasar/extras - 1.15.5 -- Quasar Framework fonts, icons and animations
  eslint-plugin-quasar - Not installed
  vue - 3.2.37 -- The progressive JavaScript framework for building modern web UI.
  vue-router - 4.1.3
  pinia - Not installed
  vuex - Not installed
  electron - Not installed
  electron-packager - Not installed
  electron-builder - Not installed
  @babel/core - 7.20.2 -- Babel compiler core.
  webpack - 5.75.0 -- Packs CommonJs/AMD modules for the browser. Allows to split your codebase into multiple bundles, which can be loaded on demand. Support loaders to preprocess files, i.e. json, jsx, es7, css, less, ... and your custom stuff.
  webpack-dev-server - 4.9.3 -- Serves a webpack app. Updates the browser on changes.
  workbox-webpack-plugin - Not installed
  register-service-worker - 1.7.2 -- Script for registering service worker, with hooks
  typescript - 4.5.5 -- TypeScript is a language for application scale JavaScript development
  @capacitor/core - Not installed
  @capacitor/cli - Not installed
  @capacitor/android - Not installed
  @capacitor/ios - Not installed

Quasar App Extensions
  *None installed*

**Comment by MonkeyFGB** [comment]
Is the 404 Not found error given by the browser, or is the 404 framework page shown?

**Reply by Cabezagithub** [reply]
404 Not Found
---
nginx/1.18.0 (Ubuntu)

It shows in the browser.

**Reply by MonkeyFGB** [reply]
Could you try to create static html file test.html and verify thats it is visible from browser?
Because it seems be a Web server issue, not application problem.

**Comment by metalsadman** [comment, marked as answer]
set your server for vue router history mode https://router.vuejs.org/guide/essentials/history-mode.html#nginx

**Reply by Cabezagithub** [reply]
Excellent Metalsadman ... good, good.
It is not an error, it is a configuration of the Web Server.
Really thank you.
Oswald
